### ISSUE TITLE
Template cache not invalidating when related entry updated

### ISSUE THREAD

**Issue by icreatestuff** [craftcms/cms#7144]
### Description
I have an entry that includes a Neo field 'Content Blocks' that contains a block 'Team' and a child block 'Team List' with an entries field 'Team Members' – see screengrab below for visual structure. If I update an entry within that entries field e.g. 'Matt Jones', the template cache for the 'Team ' page isn't invalidated. 

![Screenshot 2020-11-16 at 10 27 31](https://user-images.githubusercontent.com/1300714/99241953-739c6700-27f6-11eb-82c0-32ed6d028cd4.png)

The slimmed down template code I have for the 'team' page is

```twig
{% extends "_layouts/master" %}

{# Eager load elements #}
{% set eagerLoad = [
    'contentBlocks', 
    'contentBlocks.teamList:teamMembers',
    'contentBlocks.teamList:teamMembers.profileImage'
] %}
    
{% do craft.app.elements.eagerLoadElements(
    className(entry),
    [entry],
    eagerLoad
) %}
   
{% block content %}
    {% cache globally using key 'entry-content-' ~ entry.uid %}
        {# Page Sections! #}
        {# Preparation for the eager-loaded Neo field #}
        {% set contentBlocks = entry.contentBlocks %}

        {% for block in contentBlocks %}
            {% do block.useMemoized(contentBlocks) %}
        {% endfor %}

        {% for block in contentBlocks if block.level == 1 %}
            {% if block.type.handle == 'team' %}
                {% for childblock in block.children.all() %}
                    {% for person in childblock.teamMembers %}
                        {{ person.title }}
                    {% endfor %}
                {% endfor %}
            {% endif %}
        {% endfor %}
    {% endcache %}
{% endblock %}
```

To summarise, if I save the 'Team' entry itself it's template cache does invalidate correctly, if I update one of the related entries within the 'Team Members' field, it doesn't.

I must admit I'm not sure if this is due to the complexity of the nesting of fields and related entries, or whether it's an issue specific to Neo. 

Any help greatly appreciated. Thanks

### Additional info

- Craft version: 3.5.15.1
- PHP version: 7.3.20
- Database driver & version: MySQL 5.7.31
- Plugins & versions: Neo 2.8.14

**Comment by brandonkelly** [comment]
That’s because you are loading the block data outside of the `{% cache %}` tags, so they aren’t recorded as dependencies of the cache.

Change your template to this, and it should work as expected:

```twig
{% extends "_layouts/master" %}

{% block content %}
  {% cache globally using key 'entry-content-' ~ entry.uid %}
    {# Page Sections! #}
    {# Preparation for the eager-loaded Neo field #}
    {% set contentBlocks = clone(entry.contentBlocks)
      .with(['teamList:teamMembers.profileImage'])
      .all() %}

    {% for block in contentBlocks %}
      {% do block.useMemoized(contentBlocks) %}
    {% endfor %}

    {% for block in contentBlocks if block.level == 1 %}
      {% if block.type.handle == 'team' %}
        {% for childblock in block.children.all() %}
          {% for person in childblock.teamMembers %}
            {{ person.title }}
          {% endfor %}
        {% endfor %}
      {% endif %}
    {% endfor %}
  {% endcache %}
{% endblock %}
```

**Comment by icreatestuff** [comment]
That did the trick thanks @brandonkelly 🎉
